src/main/java/gregtech/api/recipes/builders/AssemblyLineRecipeBuilder.java

```diff
@@ -0,0 +1,150 @@
+package gregtech.api.recipes.builders;
+
+import gregtech.api.GTValues;
+import gregtech.api.recipes.Recipe;
+import gregtech.api.recipes.RecipeBuilder;
+import gregtech.api.recipes.RecipeMap;
+import gregtech.api.recipes.machines.IResearchRecipeMap;
+import gregtech.api.recipes.recipeproperties.ResearchProperty;
+import gregtech.api.util.EnumValidationResult;
+import gregtech.api.util.GTLog;
+import gregtech.api.util.ValidationResult;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+
+import javax.annotation.Nonnull;
+
+public class AssemblyLineRecipeBuilder extends RecipeBuilder<AssemblyLineRecipeBuilder> {
+
+    private ItemStack researchItem = ItemStack.EMPTY;
+    private int scanDuration = -1;
+    private int scanEUt = -1;
+    private boolean shouldAddResearchRecipe = true;
+
+    public AssemblyLineRecipeBuilder() {/**/}
+
+    @SuppressWarnings("unused")
+    public AssemblyLineRecipeBuilder(Recipe recipe, RecipeMap<AssemblyLineRecipeBuilder> recipeMap) {
+        super(recipe, recipeMap);
+    }
+
+    public AssemblyLineRecipeBuilder(AssemblyLineRecipeBuilder recipeBuilder) {
+        super(recipeBuilder);
+        this.researchItem = recipeBuilder.researchItem;
+        this.scanDuration = recipeBuilder.scanDuration;
+        this.scanEUt = recipeBuilder.scanEUt;
+        this.shouldAddResearchRecipe = recipeBuilder.shouldAddResearchRecipe;
+    }
+
+    @Override
+    public AssemblyLineRecipeBuilder copy() {
+        return new AssemblyLineRecipeBuilder(this);
+    }
+
+    @Override
+    public boolean applyPropertyCT(@Nonnull String key, Object value) {
+        if (key.equals(ResearchProperty.KEY)) {
+            this.research(value.toString());
```

Comment:
Why for CT do we want to pass a String, when we could just pass the itemStack directly, like in the Implosion Compressor Property?